### DISCUSSION TITLE
Announcing Nuekit!

### DISCUSSION THREAD

**Discussion by tipiirai** [nuejs/nue#108] [Announcements]
Nuekit is a simple alternative to Next.js and Astro with support for universal hot-reloading. A key part of the bigger picture:

<img width="973" alt="image" src="https://github.com/nuejs/nue/assets/169330/b3252396-9fb5-4f6d-b60d-87b48e36d0d1">

Check out: https://nuejs.org/blog/nuekit-010/

**Comment by seneca** [comment]
Wonderful, thank you. Expected it just before Christmas but Christmas came earlier this year.

**Comment by seneca** [comment]
Hmm, I'm not getting hot reload. I'm on a Mac, which should not make any difference. I've installed nuekit with bun. But I get on my console: Not found /@nue/diffdom.js.

**Reply by tipiirai** [reply]
Damn. I'm on Mac too so it should deinitely work there. Checking out

**Reply by seneca** [reply]
I'd like to add that I'm following the tutorial. Is there a yaml file or anything like that that should point to @nue? I can see I've got it in my .dist/dev/@nue folder so it's there.

**Reply by tipiirai** [reply]
I pushed a fix and a new nuekit version. You must upgrade to latest with `bun install nuekit@latest --global` then do `rm -r .dist` on the blog folder and restart nue with `nue`... Should work :pray:

**Comment by seneca** [comment]
Would like to report that the fix works. Thank you. :-)

**Reply by tipiirai** [reply]
Thank you! Pretty critical fix. My announcement brags so much with hot-reload

**Comment by seneca** [comment]
I'm not sure whether you've noticed that in the tutorial Build a simple blog with Nue, section Hot-reloading, you have a missing video line: ! video: completing the page the first page. I'm not sure whether this is a missing link or a reminder to yourself to do it later.

**Reply by tipiirai** [reply]
Good catch! I forgot that line in there. Planning to add a video there, but not sure if that's necessary. Cleaned up the doc now. Thanks

**Comment by seneca** [comment]
The CLI has the following sentence. It looks to me as if it has unnecessary rhe?

File matches
  Only build files that match the rhe rest of the arguments. For example:

**Reply by tipiirai** [reply]
Thanks. Fixed on [this commit](https://github.com/nuejs/nue/commit/2b9cb9520f1d9140bb07fd776d3961282fe22c19#diff-db3bfc16365772ecbf4b2b684051da16f215fa3aba29f6333c3c6b3ecf5f3450L21)

**Reply by seneca** [reply]
I think the picture in the blog will also need to be changed because it still carries that mistake there.

**Reply by tipiirai** [reply]
Sorry. Which picture? Can you give me a direct link? Thanks!

**Reply by seneca** [reply]
it's this:
https://nuejs.org/img/cli-help.png

The picture just below the first paragraph in the Command Line Interface heading.

**Comment by seneca** [comment]
What a pity we can't have bun as the officially supported package for everything. I've been trying to use it to download a template with bun create nue@latest but as the note says it just doesn't work.. It would add to the simplicity of Nue if that was possible. Resorting to other packages like pnpm adds unnecessary complexity.

**Reply by tipiirai** [reply]
Absolutely! I'm sure that is temporary. Bun is still in its early stages but proceeding fast. I'll post a bug report for this.

**Comment by seneca** [comment]
I think something is missing in the following text in Content collections, Content entries
In addition to these user-defined properties, Nue provides the following system properties for each content entry:

and no information given.

**Reply by tipiirai** [reply]
Now fixed: https://nuejs.org/docs/concepts/content-collections.html

Thanks!

**Comment by seneca** [comment]
I'm not sure whether I'm doing something wrong but I am trying to create another post in the posts folder of the blog example. I want to create it with its own layout.html, img, etc to be self contained. So the structure is: new-post->index.md, ->layout.html, ->img. But from what I can see the layout.html in the new-post folder is not taken into account when rendering the  page. It defaults to the layout.html in the posts folder above it.

**Reply by tipiirai** [reply]
This is how it works currently. But I see this coming. Will add this feature to Nuekit 0.2

**Reply by tipiirai** [reply]
Added an issue for this: https://github.com/nuejs/nue/issues/112
